### PR TITLE
Add a recipe check to MuTEs

### PR DESCRIPTION
**Pull request by BlueWeabo**
With working outputs at end of craft time!

### CODE REVIEW

File: src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java [hunk 1]
```diff
@@ -226,6 +246,16 @@ private void loadUpgradeInventoriesFromNBT(NBTTagCompound aNBT) {
         }
     }
 
+    private void loadItemsToOutput(NBTTagCompound aNBT) {
+        final NBTTagList tList = aNBT.getTagList(NBT.ITEM_OUT, 10);
+        mItemsToOutput = new ItemStack[tList.tagCount()];
+        for (int i = 0; i < tList.tagCount(); i++) {
+            final NBTTagCompound tNBT = tList.getCompoundTagAt(i);
+            final int tSlot = tNBT.getShort("s");
```
Comment (minecraft7771) [inline src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java:254]: Why not getByte here? You use setbyte to write "s".

File: src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java [hunk 2]
```diff
@@ -435,9 +465,42 @@ public void onPostTick(long aTick, boolean aIsServerSide) {
...
+                mProgressTime = 0;
+                mMaxProgressTime = 0;
+                outputItems();
+                if (isAllowedToWork()) {
+                    checkRecipe();
+                }
+            }
+        } else {
+            if (aTick % 100 == 0 || hasWorkJustBeenEnabled() || hasInventoryBeenModified()) {
+
```
Comment (minecraft7771) [inline src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java:489]: Blank line here seems unintended.

File: src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java [hunk 3]
```diff
@@ -435,9 +465,42 @@ public void onPostTick(long aTick, boolean aIsServerSide) {
...
+            if (mMaxProgressTime > 0 && ++mProgressTime >= mMaxProgressTime) {
+                mProgressTime = 0;
+                mMaxProgressTime = 0;
+                outputItems();
+                if (isAllowedToWork()) {
+                    checkRecipe();
+                }
+            }
+        } else {
+            if (aTick % 100 == 0 || hasWorkJustBeenEnabled() || hasInventoryBeenModified()) {
```
Comment (minecraft7771) [inline src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java:488]: Maybe better to make a constant for the 100.

File: src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java [hunk 4]
```diff
@@ -435,9 +465,42 @@ public void onPostTick(long aTick, boolean aIsServerSide) {
...
+                    if (checkRecipe()) {
+                        markDirty();
+                    }
+                }
+            }
+        }
+    }
+
+    protected boolean checkRecipe() {
+        boolean result = checkRecipe(null);
```
Comment (minecraft7771) [inline src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java:500]: If you do nothing with result but return it, why not inline the method call to return?

**Review by minecraft7771 — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java (hunk 1 above)
Comment (BlueWeabo) [reply src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockController.java:254]: i just copied over code. you think i write it each time?

**Review by minecraft7771 — APPROVED** [review]

**Pull request merged**